--- <CPython-3.4.9>/test/test_global.py
+++ lib/test/test_global.py
@@ -1,19 +1,16 @@
 """Verify that warnings are issued for global statements following use."""
 
-from test.support import run_unittest, check_syntax_error, check_warnings
+from test.support import run_unittest, check_syntax_error                       ###
 import unittest
-import warnings
 
+try:                                                                            ###
+    compile                                                                     ###
+except NameError:                                                               ###
+    compile = None                                                              ###
 
+@unittest.skipUnless(compile, "requires compile()")                             ###
 class GlobalTests(unittest.TestCase):
 
-    def setUp(self):
-        self._warnings_manager = check_warnings()
-        self._warnings_manager.__enter__()
-        warnings.filterwarnings("error", module="<test string>")
-
-    def tearDown(self):
-        self._warnings_manager.__exit__(None, None, None)
 
 
     def test1(self):
@@ -52,10 +49,4 @@
         compile(prog_text_4, "<test string>", "exec")
 
 
-def test_main():
-    with warnings.catch_warnings():
-        warnings.filterwarnings("error", module="<test string>")
-        run_unittest(GlobalTests)
-
-if __name__ == "__main__":
-    test_main()
+